### ISSUE TITLE
Generic attributes don't properly resolve ambiguity with verbatim identifier

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#54772]
**Version Used**: 3661e8354e3887e9826300171080c28e0d241a8e

**Steps to Reproduce**:

```cs
public class AAttribute<T> : System.Attribute {}
public class A<T> : System.Attribute {}

[@A<int>]
[AAttribute<int>]
public class C
{
}
```

**Expected Behavior**: No errors when "generic attributes" feature is enabled.

**Actual Behavior**: Ambiguity error on `[@A<int>]`.

Note that the corresponding scenario for a non-generic attribute works. The issue seems to lie in the difference between `BindNonGenericSimpleNamespaceOrTypeOrAliasSymbol` and `BindGenericSimpleNamespaceOrTypeOrAliasSymbol`. It feels like it's not a high priority scenario so punting for now.
